script_test.go

```diff
@@ -1850,6 +1850,19 @@ func TestReadReturnsErrorGivenReadErrorOnPipe(t *testing.T) {
 	}
 }
 
+// TestWithStdErrAfterExec is a regression test that was added to test against
+// a race condition for Pipe.stderr.
+func TestWithStdErrAfterExec(t *testing.T) {
```

Comment:
Testing for data races is not an exact science, but it's generally a good idea to put the doubtful code under as much stress as possible—calling it a few hundred times from two or more concurrent goroutines, for example. I don't think you mentioned whether this test failed without your fix, with or without the race detector? I wouldn't have thought it would.

Can you put together a test that fails _fairly_ reliably without the mutex lock? I don't know if you've read my book [Know Go](https://bitfieldconsulting.com/books/generics) (few people have, so I don't blame you), but there's a section in there on "smoke testing" code for concurrency safety. Here's one of the examples:

```go
s := NewSetC(1, 2, 3)
var wg sync.WaitGroup
wg.Add(1)
go func() {
	for i := range 1000 {
		s.Add(i)
	}
	wg.Done()
}()
for i := range 1000 {
	_ = s.All()
}
wg.Wait()
fmt.Println("We made it!")
// Output:
// We made it!
```

Reply:
@bitfield Thanks for the detailed comment. Do we want a test that fails without the race detector on? The current test I've written fails with the race detector on but not without.

Reply:
If it fails with the race detector on, that's good enough!